### PR TITLE
packaging/fedora, tests: Fix build for snapd and enable a test for Fedora

### PR DESCRIPTION
**Pull request by Conan-Kudo**
These changes come from pending changes for Fedora packaging for snapd 2.27.

Signed-off-by: Neal Gompa <ngompa13@gmail.com>

### CODE REVIEW

File: packaging/fedora/snapd.spec [hunk 1]
```diff
@@ -345,6 +348,10 @@ GOFLAGS="$GOFLAGS -tags withtestkeys"
 %gobuild -o bin/snapctl $GOFLAGS %{import_path}/cmd/snapctl
 %gobuild -o bin/snap-update-ns $GOFLAGS %{import_path}/cmd/snap-update-ns
 
+# We don't need mvo5 fork for seccomp, as we have seccomp 2.3.x
```
Comment (mvo5) [inline packaging/fedora/snapd.spec]: Nice!

File: packaging/fedora/snapd.spec [hunk 2]
```diff
@@ -430,12 +439,16 @@ popd
...
 pushd ./data/systemd
-%make_install SYSTEMDSYSTEMUNITDIR="%{_unitdir}"
+%make_install SYSTEMDSYSTEMUNITDIR="%{_unitdir}" BINDIR="%{_bindir}" LIBEXECDIR="%{_libexecdir}"
 # Remove snappy core specific units
 rm -fv %{buildroot}%{_unitdir}/snapd.system-shutdown.service
 rm -fv %{buildroot}%{_unitdir}/snap-repair.*
+rm -fv %{buildroot}%{_unitdir}/snapd.core-fixup.*
 popd
 
+# Rename snapd.core-fixup.sh
```
Comment (mvo5) [inline packaging/fedora/snapd.spec]: Why is this renamed? The unit file referes to it with the `.sh` suffix. Having said this, this unit is only relevant on core systems so its fine to just not package it at all on a fedora/opensuse system.

**Comment by Conan-Kudo** [comment]
@mvo5 I just purge the file now.

**Comment by chipaca** [comment]
Could you also include, in this pr, the change to `tests/main/help/task.yaml` that makes that test run on fedora?

**Comment by Conan-Kudo** [comment]
@chipaca Done.

File: packaging/fedora/snapd.spec [hunk 3]
```diff
@@ -430,12 +439,16 @@ popd
...
-%make_install SYSTEMDSYSTEMUNITDIR="%{_unitdir}"
+%make_install SYSTEMDSYSTEMUNITDIR="%{_unitdir}" BINDIR="%{_bindir}" LIBEXECDIR="%{_libexecdir}"
 # Remove snappy core specific units
 rm -fv %{buildroot}%{_unitdir}/snapd.system-shutdown.service
 rm -fv %{buildroot}%{_unitdir}/snap-repair.*
+rm -fv %{buildroot}%{_unitdir}/snapd.core-fixup.*
 popd
 
+# Remove snappy core specific scripts
+rm %{buildroot}%{_libexecdir}/snapd/snapd.core-fixup.sh
```
Comment (zyga) [inline packaging/fedora/snapd.spec]: Thanks!

File: tests/lib/pkgdb.sh [hunk 4]
```diff
@@ -279,13 +279,13 @@ distro_install_build_snapd(){
         packages=
         case "$SPREAD_SYSTEM" in
             ubuntu-*|debian-*)
-                packages="${GOHOME}"/snapd_*.deb
+                packages="${GOHOME}/snapd_*.deb"
```
Comment (zyga) [inline tests/lib/pkgdb.sh]: Hmm, does it really do the right thing? Compare `ls "$HOME/*"` and `ls "$HOME"/*`?

File: tests/lib/pkgdb.sh [hunk 5]
```diff
@@ -279,13 +279,13 @@ distro_install_build_snapd(){
...
                 ;;
             fedora-*)
-                packages="${GOHOME}"/snap-confine*.rpm "${GOPATH}"/snapd*.rpm
+                packages="${GOHOME}/snap-confine*.rpm ${GOPATH}/snapd*.rpm"
```
Comment (zyga) [inline tests/lib/pkgdb.sh]: Same here

File: tests/lib/pkgdb.sh [hunk 6]
```diff
@@ -279,13 +279,13 @@ distro_install_build_snapd(){
...
                 ;;
             opensuse-*)
-                packages="${GOHOME}"/snapd*.rpm
+                packages="${GOHOME}/snapd*.rpm"
```
Comment (zyga) [inline tests/lib/pkgdb.sh]: And here

File: tests/lib/prepare-project.sh [hunk 7]
```diff
@@ -76,7 +76,7 @@ build_rpm() {
     mkdir -p "/tmp/pkg/snapd-$version"
     cp -rav -- * "/tmp/pkg/snapd-$version/"
     mkdir -p "$rpm_dir/SOURCES"
-    (cd /tmp/pkg || tar c${archive_compression}f "$rpm_dir/SOURCES/$archive_name" "snapd-$version" "$extra_tar_args")
+    (cd /tmp/pkg && tar c${archive_compression}f "$rpm_dir/SOURCES/$archive_name" "snapd-$version" $extra_tar_args)
```
Comment (zyga) [inline tests/lib/prepare-project.sh]: Nice catch!!

File: tests/main/help/task.yaml [hunk 8]
```diff
@@ -1,4 +1,5 @@
 summary: Check commands help
+systems: [+fedora-*]
```
Comment (zyga) [inline tests/main/help/task.yaml]: :heart:

**Comment by Conan-Kudo** [comment]
@zyga Fixed.
